lang/cs/Org.Apache.REEF.IO/FileSystem/Hadoop/HadoopFileSystem.cs

```diff
@@ -37,22 +37,15 @@ internal sealed class HadoopFileSystem : IFileSystem
         private static readonly Regex NoSuchFileOrDirectoryRegEx = new Regex("^ls: `.*': No such file or directory");
         private static readonly Regex LsFirstLineRegex = new Regex("^Found .* items");
         private readonly HdfsCommandRunner _commandRunner;
-        private readonly string _uriPrefix;
 
         [Inject]
         private HadoopFileSystem(HdfsCommandRunner commandRunner)
         {
             _commandRunner = commandRunner;
-            _uriPrefix = GetUriPrefix();
+            UriPrefix = GetUriPrefix();
         }
 
-        /// <summary>
-        /// The Prefix used for URIs on this FileSystem.
-        /// </summary>
-        public string UriPrefix
-        {
-            get { return _uriPrefix; }
-        }
+        public string UriPrefix { get; private set; }
```

Comment:
why the private setter? It makes the field mutable, and immutable would be better from a thread safety perspective.